src/common/rbac.ts

```diff
@@ -7,38 +7,49 @@ export type KubeResource =
 }
 
 // TODO: auto-populate all resources dynamically (see: kubectl api-resources -o=wide -v=7)
 export const apiResources: KubeApiResource[] = [
-  { resource: "configmaps" },
-  { resource: "cronjobs", group: "batch" },
-  { resource: "customresourcedefinitions", group: "apiextensions.k8s.io" },
-  { resource: "daemonsets", group: "apps" },
-  { resource: "deployments", group: "apps" },
-  { resource: "endpoints" },
-  { resource: "events" },
-  { resource: "horizontalpodautoscalers" },
-  { resource: "ingresses", group: "networking.k8s.io" },
-  { resource: "jobs", group: "batch" },
-  { resource: "namespaces" },
-  { resource: "networkpolicies", group: "networking.k8s.io" },
-  { resource: "nodes" },
-  { resource: "persistentvolumes" },
-  { resource: "persistentvolumeclaims" },
-  { resource: "pods" },
-  { resource: "poddisruptionbudgets" },
-  { resource: "podsecuritypolicies" },
-  { resource: "resourcequotas" },
-  { resource: "replicasets", group: "apps" },
-  { resource: "secrets" },
-  { resource: "services" },
-  { resource: "statefulsets", group: "apps" },
-  { resource: "storageclasses", group: "storage.k8s.io" },
+  { kind: "ConfigMap", resource: "configmaps" },
+  { kind: "CronJob", resource: "cronjobs", group: "batch" },
+  { kind: "CustomResourceDefinition", resource: "customresourcedefinitions", group: "apiextensions.k8s.io" },
+  { kind: "DaemonSet", resource: "daemonsets", group: "apps" },
+  { kind: "Deployment", resource: "deployments", group: "apps" },
+  { kind: "Endpoint", resource: "endpoints" },
+  { kind: "Event", resource: "events" },
+  { kind: "HorizontalPodAutoscaler", resource: "horizontalpodautoscalers" },
+  { kind: "Ingress", resource: "ingresses", group: "networking.k8s.io" },
+  { kind: "Job", resource: "jobs", group: "batch" },
+  { kind: "Namespace", resource: "namespaces" },
+  { kind: "NetworkPolicy", resource: "networkpolicies", group: "networking.k8s.io" },
+  { kind: "Node", resource: "nodes" },
+  { kind: "PersistentVolume", resource: "persistentvolumes" },
+  { kind: "PersistentVolumeClaim", resource: "persistentvolumeclaims" },
+  { kind: "Pod", resource: "pods" },
+  { kind: "PodDisruptionBudget", resource: "poddisruptionbudgets" },
+  { kind: "PodSecurityPolicy", resource: "podsecuritypolicies" },
+  { kind: "ResourceQuota", resource: "resourcequotas" },
+  { kind: "ReplicaSet", resource: "replicasets", group: "apps" },
+  { kind: "Secret", resource: "secrets" },
+  { kind: "Service", resource: "services" },
+  { kind: "StatefulSet", resource: "statefulsets", group: "apps" },
+  { kind: "StorageClass", resource: "storageclasses", group: "storage.k8s.io" },
 ];
 
+export function isAllowedResourceType(kind: string): boolean {
+  const apiResource = apiResources.find(resource => resource.kind === kind);
+
+  if (apiResource) {
+    return getHostedCluster().allowedResources.includes(apiResource.resource);
+  }
+
+  return true; // allowed by default for other resources
+}
+
```

Comment:
Why is this not a method on `Cluster`? Then the consumer can decide where the cluster comes from. For instance `getHostedCluster().isAlloedResourceType(kind)`

Reply:
Because https://github.com/lensapp/lens/pull/1918#discussion_r556272697

Reply:
No, this function uses `getHostedCluster()` internally so it is no better (imo).

Reply:
Ok, anyway could be added in separated PR as needed.

Reply:
I don't see why it should wait for a followup PR. This work is being done in this PR and as Jari noted we should be moving away from `getHostedCluster`.

Reply:
Alright! 👨🏼‍💻

Reply:
Added new methods: 

- cluster.isAllowedResource(kind: string): boolean
- async store.resolveCluster(): Cluster

Anyway.. `getHostedCluster()` dynamically imported in `resolveCluster` to get access to `isAllowedResource`, but now if store is used outside of cluster frame end user can override method and provide own cluster in different env.